music_this_week_app/backend/spotifyHandler.py

```diff
@@ -161,14 +192,16 @@ def get_song_list(self, artist_URIs, N=99, order="shuffled"):
         """
 
         # Calculate number of tracks per artist. Round up to nearest int w/ int division then trim list later.
-        number_of_tracks_per_artist = N // len(artist_URIs) + 1
+        number_of_tracks_per_artist = N // self.artists_uris_found + 1
         if number_of_tracks_per_artist > 10:
             print("Number of tracks per artist, %i, cannot be greater than 10." %number_of_tracks_per_artist)
 
         # Identify songs for the playlist; list of track URIs
         tracks = []
         for a in artist_URIs:
             tracks = tracks + self.find_top_tracks(a, N=number_of_tracks_per_artist)
+            self.save_new_artist(self.artists_to_save[a], a, tracks)
```

Comment:
Maybe it would be better to have separate functions for separate functionality.

With that name save_new_artist should create a new entry in the database (and do nothing else)
You could create another function to add more info to an existing entry in the database